librad/src/net/protocol.rs

```diff
@@ -174,6 +174,34 @@ impl TinCans {
         rx.await.unwrap_or_default()
     }
 
+    pub async fn membership(&self) -> event::downstream::MembershipInfo {
```

Comment:
My only critique is that this is pretty much the same as the above `connected_peers` function. It could be useful to just abstract it out and then it's:
* Build channel
* Send message
* Expect response + use default otherwise

Might make it easier to add any future cases. Lemme know what yis think

Reply:
Actually... `connected_peers` is subtly different: it considers _all_ connections. It is permissible to talk to a node but not participate in gossip (eg. only `git`, or some to-come info API).

Did someone say write more docs?
![image](https://user-images.githubusercontent.com/6163/109834407-6dfe4c00-7c42-11eb-8336-f42b132c3880.png)

Reply:
While with this addition the rule of 3 applies, I'm not sure how much we gain from the DRYness here.

Reply:
Can we leave this for a future exercise?

Reply:
Does that only differ in the receiver of `ConnectedPeers`? When I diff these two function blocks:

```
5c4
<     self.downstream.send(Downstream::Info(Membership(tx)))
---
>     self.downstream.send(Downstream::Info(ConnectedPeers(tx)))
8c7
<         Downstream::Info(Membership(reply)) => {
---
>         Downstream::Info(ConnectedPeers(reply)) => {
13c12
<                 .send(MembershipInfo::default())
---
>                 .send(vec![])
19d17
< 
```

Reply:
Oh code structure. I hear the preferred way to do this in Rust is to make a macro ... ahm wait

:P

Reply:
I'm fine with how it is